apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php

```diff
@@ -560,3 +554,62 @@ function wpcom_is_previewing_global_styles( ?int $user_id = null ) {
 	// phpcs:ignore WordPress.Security.NonceVerification.Recommended
 	return ! isset( $_GET['hide-global-styles'] ) && user_can( $user_id, 'administrator' );
 }
+
+/**
+ * Checks whether the site has a plan that grants access to the Global Styles feature.
+ *
+ * @param  int $blog_id Blog ID.
+ * @return bool Whether the site has access to Global Styles.
+ */
+function wpcom_site_has_global_styles_feature( $blog_id = 0 ) {
+	/*
+	 * Non-Simple sites on a lower plan are temporary edge cases. We grant them access
+	 * to Global Styles to prevent unexpected temporary changes in their styles.
+	 */
+	if ( ! defined( 'IS_WPCOM' ) || ! IS_WPCOM ) {
+		return true;
+	}
+
+	if ( wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id ) ) {
+		return true;
+	}
+
+	// Limited Global Styles on Personal A/B test.
+	$owner_id = wpcom_get_blog_owner( $blog_id );
+	if ( ! $owner_id ) {
+		return false;
+	}
+
+	$owner = get_userdata( $owner_id );
+	if ( ! $owner ) {
+		return false;
+	}
+
+	if ( 'treatment' === \ExPlat\get_user_assignment( 'calypso_global_styles_personal', $owner ) ) {
```

Comment:
We don't need to run this if `wpcom_global_styles_has_blog_sticker( 'wpcom-global-styles-personal-plan' );` is already true, which will reduce queries a little.

Reply:
Solved in https://github.com/Automattic/wp-calypso/commit/4b76a9af82d6c07e578c4b9388c7c86d2de96133!